tests/calico_cni_k8s_test.go

```diff
@@ -2210,6 +2361,9 @@ var _ = Describe("Kubernetes CNI tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 			containerID2 := "random-cid"
 			defer func() {
+				// Delete pod
+				ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name2)
```

Comment:
I think if we're going to this we need to make sure it occurs _after_ we call the CNI plugin (e.g., `DeleteContainerWithId` below).